lib/ansible/modules/cloud/ovirt/ovirt_host_networks.py

```diff
@@ -160,48 +175,54 @@
 )
 
 
-def get_mode_type(mode_number):
-    """
-    Adaptive transmit load balancing (balance-tlb): mode=1 miimon=100
-    Dynamic link aggregation (802.3ad): mode=2 miimon=100
-    Load balance (balance-xor): mode=3 miimon=100
-    Active-Backup: mode=4 miimon=100 xmit_hash_policy=2
-    """
+def get_bond_options(mode, usr_opts):
+    MIIMON_100 = dict(miimon='100')
+    DEFAULT_MODE_OPTS = {
+        '1': MIIMON_100,
+        '2': MIIMON_100,
+        '3': MIIMON_100,
+        '4': dict(xmit_hash_policy='2', **MIIMON_100)
+    }
+
     options = []
-    if mode_number is None:
+    if mode is None:
         return options
 
     def get_type_name(mode_number):
         """
         We need to maintain this type strings, for the __compare_options method,
         for easier comparision.
         """
-        return [
+        modes = [
             'Active-Backup',
             'Load balance (balance-xor)',
             None,
             'Dynamic link aggregation (802.3ad)',
-        ][mode_number]
+        ]
+        if (not 0 < mode_number <= len(modes) - 1):
+            return None
+        return modes[mode_number - 1]
 
     try:
-        mode_number = int(mode_number)
-        if mode_number >= 1 and mode_number <= 4:
-            if mode_number == 4:
-                options.append(otypes.Option(name='xmit_hash_policy', value='2'))
-
-            options.append(otypes.Option(name='miimon', value='100'))
-            options.append(
-                otypes.Option(
-                    name='mode',
-                    type=get_type_name(mode_number - 1),
-                    value=str(mode_number)
-                )
-            )
-        else:
-            options.append(otypes.Option(name='mode', value=str(mode_number)))
+        mode_number = int(mode)
     except ValueError:
-        raise Exception("Bond mode must be a number.")
+        raise Exception('Bond mode must be a number.')
+
+    options.append(
+        otypes.Option(
+            name='mode',
+            type=get_type_name(mode_number),
+            value=str(mode_number)
+        )
+    )
 
+    opts_dict = DEFAULT_MODE_OPTS.get(mode, {})
+    opts_dict.update(**usr_opts)
+
+    options.extend(
+        [otypes.Option(name=opt, value=value)
```

Comment:
I am afraid this won't work in compare method bceause you don't pass `type`, but API does fill it, can you please verify idepmtency works correctly?

Reply:
We've been only passing type for "mode", and I do that still (line#214). I verified this locally.

Reply:
Oh I see :)